### ISSUE TITLE
BadState with null safety version

### ISSUE THREAD

**Issue by bsutton** [google/crclib.dart#15]
I'm trying to run with the nullsaftey version but I'm getting bad state error when I call single.

```
Future<T> get first {
    _Future<T> future = new _Future<T>();
    StreamSubscription<T> subscription =
        this.listen(null, onError: future._completeError, onDone: () {
      try {
throw IterableElementError.noElement();

StateError (Bad state: No element)
```

This is my code:
```dart
import 'dart:io';

import 'package:crclib/catalog.dart';
import 'package:dcli/dcli.dart';
import 'package:test/test.dart';

void main() {
  
  test('hash performance', () async {
    withTempFile((largeFile) async {
      try {
        createLargeFile(largeFile);
        now();
        await File(largeFile).openRead().transform(Crc32()).single;
        now();
      } catch (e) {
        print(e);
      }
    });
  });
}

void now() {
  print(DateTime.now());
}

void createLargeFile(String largeFile) {
  for (int i = 0; i < 100000; i++) {
    largeFile.append('*' * 1000);
  }
}

```

```yaml
name: batman
version: 1.0.2
homepage: https://github.com/noojee/batman
description: A System Integrity Monitor for PCI DSS compliance. batman provides file integrity monitoring and log scanners with a built in scheduler suitable for use in a docker container.
repository: https://github.com/noojee/batman
environment: 
  sdk: '>=2.14.4 <3.0.0'
dependencies: 
  dcli: ^1.13.0
  dcli_core: ^0.0.2
  crclib:
    git:
      url: https://github.com/google/crclib.dart.git
      ref: nullsafety  
```

**Comment by postmasters** [comment]
What if you remove the `.transform(Crc32())` portion, leaving behind just `.single` right after `openRead()`? Will that also cause error? My suspicion here is in the fact that you are writing to a file without closing, and opening that file again. This is highly dependent on the filesystem in use.

**Comment by bsutton** [comment]
I've just ran this with the released 3.0.0 package and the problem has gone away.

So will close for now.
